src/grt/src/fastroute/src/utility.cpp

```diff
@@ -2302,6 +2304,107 @@ int FastRouteCore::edgeShiftNew(Tree& t, int net)
   return (numShift);
 }
 
+odb::Rect FastRouteCore::globalRoutingToBox(const GSegment& route)
+{
+  int init_x, init_y;
+  int final_x, final_y;
+
+  if (route.init_x < route.final_x) {
+    init_x = route.init_x;
+    final_x = route.final_x;
+  } else {
+    init_x = route.final_x;
+    final_x = route.init_x;
+  }
+
+  if (route.init_y < route.final_y) {
+    init_y = route.init_y;
+    final_y = route.final_y;
+  } else {
+    init_y = route.final_y;
+    final_y = route.init_y;
+  }
+
+  int llX = init_x - (tile_size_ / 2);
+  int llY = init_y - (tile_size_ / 2);
+
+  int urX = final_x + (tile_size_ / 2);
+  int urY = final_y + (tile_size_ / 2);
+
+  if ((x_grid_max_ - urX) / tile_size_ < 1) {
+    urX = x_grid_max_;
+  }
```

Comment:
Is this just
```
urX = std::min(urX, x_grid_max_);
```
or is it something more complex?  Its a bit hard to follow. 

Why is nothing similar needed with llX?

Reply:
I used the same GlobalRouter::globalRoutingToBox operation, 
I think it's when the last tile is less width than the `tile_size`

Reply:
Yes, this is checking if the last tile is smaller than tile_size, and creating a larger guide for the last row/column of the route grid.